examples/src/main/java/kafka/examples/ExactlyOnceMessageProcessor.java

```diff
@@ -155,12 +157,19 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
                     // Finish the transaction. All sent records should be visible for consumption now.
                     producer.commitTransaction();
                     messageProcessed += records.count();
-                } catch (CommitFailedException e) {
-                    // In case of a retriable exception, suggest aborting the ongoing transaction for correctness.
-                    abortPreviousTransaction = true;
-                } catch (ProducerFencedException | FencedInstanceIdException e) {
-                    throw new KafkaException("Encountered fatal error during processing: " + e.getMessage());
                 }
+            } catch (CommitFailedException | InvalidOffsetException e) {
+                // In case of a retriable exception, suggest aborting the ongoing transaction for correctness.
+                // Note that abort transaction call could also throw fatal exceptions such as producer fenced.
+                producer.abortTransaction();
+
+                // The consumer fetch position also needs to be reset.
+                resetToLastCommittedPositions(consumer);
+            } catch (ProducerFencedException | FencedInstanceIdException | AuthorizationException |
```

Comment:
For the purpose of understanding EOS, the main exceptions that are worth calling out are `ProducerFencedException` and `FencedInstanceIdException`. I was suggest we write the example like this:
```java
try {
   ...
   producer.commitTransaction;
} catch (ProducerFencedException e) {
  throw KafkaException("The transactional.id $transactionalId has been claimed by another process");
} catch (FencedInstanceIdException e) {
  throw KafkaException("The group.instance.id $instanceId has been claimed by another process");
} catch (KafkaException e) {
  // If we have not been fenced, try to abort the transaction and continue. This will raise immediately
  // if the producer has hit a fatal error.
  producer.abortTransaction();
}
```